vcd/resource_vcd_vapp_firewall_rules_test.go

```diff
@@ -0,0 +1,473 @@
+// +build functional gateway ALL
+
+package vcd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccVcdVappFirewallRules(t *testing.T) {
+	if testConfig.Networking.EdgeGateway == "" {
+		t.Skip("Variable testConfig.Networking.EdgeGateway must be configured")
+		return
+	}
+
+	var (
+		vmName1  = "TestAccVcdVappFirewallRulesVm1"
+		vmName2  = "TestAccVcdVappFirewallRulesVm2"
+		vmName3  = "TestAccVcdVappFirewallRulesVm3"
+		vappName = t.Name() + "_vapp"
+	)
+
+	var params = StringMap{
+		"Org":           testConfig.VCD.Org,
+		"Vdc":           testConfig.VCD.Vdc,
+		"EdgeGateway":   testConfig.Networking.EdgeGateway,
+		"DefaultAction": "drop",
+		"ResourceName":  t.Name(),
+		"FuncName":      t.Name(),
+		"Description1":  "description1",
+		"Description2":  "description2",
+		"Description3":  "description3",
+		"Description4":  "description4",
+		"NetworkName":   "TestAccVcdVAppVmNet",
+		"VappName":      vappName,
+		"VmName1":       vmName1,
+		"VmName2":       vmName2,
+		"VmName3":       vmName3,
+	}
+	configText := templateFill(testAccVcdVappFirewallRules_rules, params)
+	params["FuncName"] = t.Name() + "-step2"
+	configTextForUpdate := templateFill(testAccVcdVappFirewallRules_rules_forUpdate, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVappFirewallRulesExists("vcd_vapp_firewall_rules."+t.Name(), params["Description1"].(string)),
```

Comment:
Please make a variable for `"vcd_vapp_firewall_rules."+t.Name()` and use it in the lines below

Reply:
refactored